massa-execution-worker/src/execution.rs

```diff
@@ -1011,6 +1017,13 @@ impl ExecutionState {
         // TODO ensure that speculative things are reset after every execution ends (incl. on error and readonly)
         // otherwise, on prod stats accumulation etc... from the API we might be counting the remainder of this speculative execution
 
+        if req.max_gas > self.config.max_gas_per_execution {
+            return Err(ExecutionError::ExecutionGasOverThreshold(format!(
+                "execution gas for read-only call is {} which is above the maximum allowed {}",
+                req.max_gas, self.config.max_gas_per_execution
+            )));
+        }
```

Comment:
this is the only useful one. All the other ones are redundant

Reply:
Also, let's rename `self.config.max_gas_per_execution` into `self.config.max_read_only_gas`